buildSrc/src/main/java/xyz/wagyourtail/doclet/webdoclet/parsers/ClassParser.java

```diff
@@ -467,7 +467,14 @@ private XMLBuilder parseType(TypeMirror type) {
                 if (params != null && !params.isEmpty()) {
                     builder.append("<");
                     for (TypeMirror param : params) {
-                        builder.append(parseType(param), ", ");
+                        if (param instanceof TypeVariable typeVariable && typeVariable.getUpperBound().equals(type)) {
+                            builder.append(typeLink = new XMLBuilder("p", true));
+                            typeLink.setClass("type primitiveType");
+                            typeLink.append(typeVariable.asElement().getSimpleName());
+                            builder.append(", ");
+                        } else {
+                            builder.append(parseType(param), ", ");
+                        }
```

Comment:
The change in the python ClassParser is for 2 occurences in which I return an array of Minecraft classes which causes the parser to crash because the array has no simple name. Instead I just use the simple name of the inner type of the array, which would fail more more dimensional arrays, but the parser stuff is already cursed.

The change in the webdoclet ClassParser is to fix recursion for generic declarations, see AbstractWidgetBuilder for example. This fix just checks if the generic parameter is equal to the generic value itself. If so just replace the generic parameter with it's name, otherwise use the old code to recursively get the generic parameters.